File: algorithms/ar-lvwbintr/python3/leftviewbt.py
class Node:
    def __init__(self, data, left=None, right=None):
        self.data = data
        self.left = left
        self.right = right


def left_view_iterative(root_node):
    if not root_node:
        return None

    left_view = []

    queue = [root_node]
    level = max_level = 0

    while len(queue):
        level_count = len(queue)
        level += 1

        for __ in range(level_count):
            node = queue.pop(0)

            if level > max_level:
                left_view.append(node.data)
                max_level = level

            if node.left:
                queue.append(node.left)
            if node.right:
                queue.append(node.right)

    return left_view


if __name__ == "__main__":
    # Build tree
    root = Node(1)
    root.left = Node(2)
    root.right = Node(3)
    #root.left.left = Node(4)
    root.left.right = Node(5)
    root.right.left = Node(6)
    root.right.right = Node(7)
    root.right.right.right = Node(8)
    root.right.right.right.left = Node(9)

    # Print left view binary tree
    print(left_view_iterative(root))

File: algorithms/ar-lvwbintr/C/leftviewbt.c
// C program to print left view of Binary Tree
#include <stdio.h>
#include <stdlib.h>

struct node {
	int data;
	struct node *left, *right;
};

// A utility function to create a new Binary Tree node
struct node* newNode(int item)
{
	struct node* temp
		= (struct node*)malloc(sizeof(struct node));
	temp->data = item;
	temp->left = temp->right = NULL;
	return temp;
}

// Recursive function to print left view of a binary tree.
void leftViewUtil(struct node* root, int level,
				int* max_level)
{
	// Base Case
	if (root == NULL)
		return;

	// If this is the first node of its level
	if (*max_level < level) {
		printf("%d\t", root->data);
		*max_level = level;
	}

	// Recur for left and right subtrees
	leftViewUtil(root->left, level + 1, max_level);
	leftViewUtil(root->right, level + 1, max_level);
}

// A wrapper over leftViewUtil()
void leftView(struct node* root)
{
	int max_level = 0;
	leftViewUtil(root, 1, &max_level);
}

// Driver code
int main()
{
	struct node* root = newNode(10);
	root->left = newNode(2);
	root->right = newNode(3);
	root->left->left = newNode(7);
	root->left->right = newNode(8);
	root->right->right = newNode(15);
	root->right->left = newNode(12);
	root->right->right->left = newNode(14);

	leftView(root);

	return 0;
}
